sdks/python/apache_beam/io/external/snowflake.py

```diff
@@ -0,0 +1,144 @@
+        ('o_auth_token', typing.Optional[unicode]),
+        ('table', typing.Optional[unicode]),
+        ('query', typing.Optional[unicode]),
+    ])
+
+
+class ReadFromSnowflake(beam.PTransform):
+  """An external PTransform which reads from Snowflake."""
+
+  URN = 'beam:external:java:snowflake:read:v1'
+
+  def __init__(
+      self,
+      server_name,
+      schema,
+      database,
+      staging_bucket_name,
+      storage_integration_name,
+      csv_mapper,
+      username=None,
+      password=None,
+      private_key_path=None,
+      private_key_passphrase=None,
+      o_auth_token=None,
+      table=None,
+      query=None,
+      expansion_service=None):
+    """
+    Initializes a read operation from Snowflake.
+
+    Required parameters:
+    :param server_name: full Snowflake server name with the following format
+        account.region.gcp.snowflakecomputing.com.
+    :param schema: name of the Snowflake schema in the database to use.
+    :param database: name of the Snowflake database to use.
+    :param staging_bucket_name: name of the Google Cloud Storage bucket.
+        Bucket will be used as a temporary location for storing CSV files.
+        Those temporary directories will be named
+        `sf_copy_csv_DATE_TIME_RANDOMSUFFIX`
+        and they will be removed automatically once Read operation finishes.
+    :param storage_integration_name: is the name of storage integration
+        object created according to Snowflake documentation.
+    :param csv_mapper: specifies a function which must translate
+        user-defined object to array of strings.
+        SnowflakeIO uses a COPY INTO <location> statement to
+        move data from a Snowflake table to Google Cloud Storage as CSV files.
+        These files are then downloaded via FileIO and processed line by line.
+        Each line is split into an array of Strings using the OpenCSV
+        The csv_mapper function job is to give the user the possibility to
+        convert the array of Strings to a user-defined type,
+        ie. GenericRecord for Avro or Parquet files, or custom objects.
+            Example:
+                ```
+                    def csv_mapper(strings_array):
+ 		                return User(strings_array[0], int(strings_array[1])))
+                ```
+    :param table or query: specifies a Snowflake table name or custom SQL query
+    :param expansion_service: specifies URL of expansion service.
+
+    Authentication parameters:
+    It's required to pass one of the following combinations of valid parameters:
+    :param username and password: specifies username and password
+        for username/password authentication method.
+    :param private_key_path and private_key_passphrase:
+        specifies a private key file and password
+        for key/ pair authentication method.
+    :param o_auth_token: specifies access token for OAuth authentication method.
+    """
+
+    self.params = ReadFromSnowflakeSchema(
```

Comment:
Have you tested this against portable Flink/Spark and or Dataflow. Please mention here the runners this have been tested for and supported. Also mention details about prerequisites for the user. See following for an example.
https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/kafka.py#L18

Reply:
I tested it against Flink. 
I added missing information.

Reply:
Can you try against Dataflow as well ? Instructions are here: https://github.com/apache/beam/tree/master/sdks/python/apache_beam/examples/kafkataxi
(it's for Kafka but you should be able to adapt)

Reply:
@chamikaramj, I tried to reproduce it for Snowflake but I'm not able to overcome the following error:

`NotImplementedError: Execution of [<ExternalTransform(PTransform) label=[ExternalTransform(beam:external:java:snowflake:read:v1)]>] not implemented in runner <apache_beam.runners.dataflow.dataflow_runner.DataflowRunner `

Do you have some idea how to solve this?

Reply:
I added instructions for running Kafka against Dataflow for HEAD to here: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/kafkataxi/README.md

For example, you need to setup "--experiments=use_runner_v2" and "--sdk_harness_container_image_overrides". Feel free to try this out and update the pydoc above if successful.

Reply:
@chamikaramj . I'm getting following error:
```
    parameter_type, constructor = cls._known_urns[fn_proto.urn]
KeyError: 'beam:window_fn:serialized_java:v1'
```

Reply:
Thanks. This is potentially due to https://issues.apache.org/jira/browse/BEAM-10507.